components/support/utils/src/main/java/mozilla/components/support/utils/URLStringUtils.kt

```diff
@@ -28,4 +30,20 @@ object URLStringUtils {
         }
         return uri.toString()
     }
+
+    private val isURLLenient by lazy {
+        // Be lenient about what is classified as potentially a URL. Anything that contains a :,
+        // ://, or . and has no internal spaces is potentially a URL.
+        //
+        // Use java.util.regex because
+        // of its explicit unicode support when using the UNICODE_CHARACTER_CLASS flag
+        // (https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html).
```

Comment:
nit: should we link to Android docs instead?

Reply:
I think that we should link to the Java documentation since we are specifically referring to the capability of the SDK. But, I see your point: keeping up to date with the versions of the SDK may be hard. I honestly don't care which one we do. Do you feel strongly one way or another?

Reply:
The Android framework contains a copy of the SDK and, for all we know, may not be representative of the SDK (for example, the behavior around unicode). Therefore, I strongly prefer to reference the Android documentation.